cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -348,43 +351,43 @@ private void processCapacityEntries(CruiseControlSpec spec, Set<NodeRef> kafkaBr
-                        outboundNetwork = processOutboundNetwork(brokerCapacity, override);
-                        for (int id : ids) {
-                            if (id == BrokerCapacity.DEFAULT_BROKER_ID) {
-                                LOGGER.warnCr(reconciliation, "Ignoring broker capacity override with illegal broker id -1.");
-                            } else {
-                                if (capacityEntries.containsKey(id)) {
-                                    if (overrideIds.add(id)) {
-                                        BrokerCapacity brokerCapacityEntry = capacityEntries.get(id);
-                                        brokerCapacityEntry.setCpu(processCpu(brokerCapacity, override, capacityEntries.get(id).getCpu()));
-                                        brokerCapacityEntry.setInboundNetwork(inboundNetwork);
-                                        brokerCapacityEntry.setOutboundNetwork(outboundNetwork);
+            if (brokerCapacity != null) {
+                // For checking for duplicate brokerIds
+                Set<Integer> overrideIds = new HashSet<>();
+                List<BrokerCapacityOverride> overrides = brokerCapacity.getOverrides();
+                // Override broker entries
+                if (overrides != null) {
+                    if (overrides.isEmpty()) {
+                        LOGGER.warnCr(reconciliation, "Ignoring empty overrides list");
+                    } else {
+                        for (BrokerCapacityOverride override : overrides) {
+                            List<Integer> ids = override.getBrokers();
+                            inboundNetwork = processInboundNetwork(brokerCapacity, override);
+                            outboundNetwork = processOutboundNetwork(brokerCapacity, override);
+                            for (int id : ids) {
+                                if (id == BrokerCapacity.DEFAULT_BROKER_ID) {
+                                    LOGGER.warnCr(reconciliation, "Ignoring broker capacity override with illegal broker id -1.");
+                                } else {
+                                    if (capacityEntries.containsKey(id)) {
+                                        if (overrideIds.add(id)) {
+                                            BrokerCapacity brokerCapacityEntry = capacityEntries.get(id);
+                                            brokerCapacityEntry.setCpu(processCpu(override, brokerCapacity, kafkaBrokerResources.get(Integer.toString(id))));
+                                            brokerCapacityEntry.setInboundNetwork(inboundNetwork);
+                                            brokerCapacityEntry.setOutboundNetwork(outboundNetwork);
+                                        } else {
+                                            LOGGER.warnCr(reconciliation, "Duplicate broker id {} found in overrides, using first occurrence.", id);
+                                        }
```

Comment:
@kyguy The block moved into a for-loop now. He has to use the correct indentation. (because it is the right thing to do but also because checkstyle should enforce it)

Reply:
As Jakub said. The block was moved inside the loop.